### PR TITLE
CEP-397: Add Compute API and response limit

### PR DESCRIPTION
**Pull request by dlazarenko-macrometa**
The PR the same as https://github.com/Macrometacorp/c84j/pull/75 but for `master`

Added methods for Compute API
- Fetches a list of all functions. `getFunctions()`
- Get information about function worker. `getInfo(final String name)`
- Fetch information about a edge worker. `getMetadata()`
- Execute a function worker. `executeFunction(String name, Map<String, Object> arguments)`
- added response limit with default value is 1MB.

### CODE REVIEW

**Review by StoyanTc — APPROVED** [review]

File: src/main/java/com/c8db/internal/InternalC8Compute.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                if (result.isArray()) {
+                    return util().deserialize(result, new Type<Collection<Map<String, Object>>>(){}.getType());
+                } else {
+                    return Arrays.asList(util().deserialize(result, new Type<Map<String, Object>>(){}.getType()));
+                }
+            }
+        };
+    }
+
+}
```
Comment (praminda-macrometa) [inline src/main/java/com/c8db/internal/InternalC8Compute.java:111]:
```suggestion
}

```

File: src/main/java/com/c8db/internal/http/HttpRequestRetryHandler.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.Arrays;
+
+public class HttpRequestRetryHandler extends DefaultHttpRequestRetryHandler {
+
+    public HttpRequestRetryHandler() {
+        super(3, false,
+            Arrays.asList(InterruptedIOException.class, UnknownHostException.class,
+                ConnectException.class, SSLException.class, MessageConstraintException.class));
+    }
+}
```
Comment (praminda-macrometa) [inline src/main/java/com/c8db/internal/http/HttpRequestRetryHandler.java:22]:
```suggestion
}

```

File: src/main/java/com/c8db/model/FxReadOptions.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  *
+  * @param type - instance of `FxType` class.
+  * @return
+  */
+ public FxReadOptions type(final FxType type) {
+  this.type = type;
+  return this;
+ }
+
+}
```
[the remark this answers is not in the document]
Comment (dlazarenko-macrometa) [reply src/main/java/com/c8db/model/FxReadOptions.java]: Done

File: src/main/java/com/c8db/internal/http/HttpRequestRetryHandler.java (hunk 2 above)
Comment (dlazarenko-macrometa) [reply src/main/java/com/c8db/internal/http/HttpRequestRetryHandler.java]: Done

File: src/main/java/com/c8db/internal/InternalC8Compute.java (hunk 1 above)
Comment (dlazarenko-macrometa) [reply src/main/java/com/c8db/internal/InternalC8Compute.java]: Done

File: src/main/java/com/c8db/entity/FxType.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * Class describes type of function worker.
+ */
+public enum FxType {
+
+    ALL,
+    AKAMAI,
+    MACROMETA
+
+}
```
[the remark this answers is not in the document]
Comment (dlazarenko-macrometa) [reply src/main/java/com/c8db/entity/FxType.java]: Done

File: src/main/java/com/c8db/entity/FxMetadataEntity.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    public String getResourceTierId() {
+        return resourceTierId;
+    }
+
+    public FxType getType() {
+        return type;
+    }
+}
```
[the remark this answers is not in the document]
Comment (dlazarenko-macrometa) [reply src/main/java/com/c8db/entity/FxMetadataEntity.java]: Done

File: src/main/java/com/c8db/entity/FxEntity.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    public FxType getType() {
+        return type;
+    }
+
+    public String getUrl() {
+        return url;
+    }
+}
```
[the remark this answers is not in the document]
Comment (dlazarenko-macrometa) [reply src/main/java/com/c8db/entity/FxEntity.java]: Done

File: src/main/java/com/c8db/C8ComputeImpl.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public FxMetadataEntity getMetadata() throws C8DBException {
+        return executor.execute(getMetadataRequest(), getMetadataResponseDeserializer(), null, Service.C8FUNCTION);
+    }
+
+    @Override
+    public Collection<Map<String, Object>> executeFunction(String name, Map<String, Object> arguments) throws C8DBException {
+        return executor.execute(executeFunctionRequest(name, arguments), executeFunctionResponseDeserializer(), null, Service.C8FUNCTION);
+    }
+}
```
[the remark this answers is not in the document]
Comment (dlazarenko-macrometa) [reply src/main/java/com/c8db/C8ComputeImpl.java]: Done

File: src/main/java/com/c8db/C8Compute.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * Execute a function worker.
+     *
+     * @param name - name of function
+     * @param arguments - set parameters with arguments as a map
+     * @return - result of execution of a function.
+     * @throws C8DBException
+     */
+    Collection<Map<String, Object>> executeFunction(String name, Map<String, Object> arguments) throws C8DBException;
+}
```
[the remark this answers is not in the document]
Comment (dlazarenko-macrometa) [reply src/main/java/com/c8db/C8Compute.java]: Done

[the remark this answers is not in the document]
Comment (dlazarenko-macrometa) [reply src/main/java/com/c8db/C8Compute.java]: Done

**Pull request merged**